characteristic_readwrite/web_app.js

```diff
@@ -0,0 +1,104 @@
+/**
+ *  Copyright 2021 Google LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+// Espruino devices publish a UART service by default.
+const nordicUARTService = '6e400001-b5a3-f393-e0a9-e50e24dcca9e';
+// The test service defined in device_code.js
+const testService = '1c84b596-222c-11eb-adc1-0242ac120002';
+// The test characteristic defined in device_code.js
+const testCharacteristic = '1c84b91a-222c-11eb-adc1-0242ac120002';
+
+/**
+ * Load the device code to the Espruino IDE.
+ */
+function loadEspruinoDeviceCode() {
+  fetch('device_code.js').then(response => response.text()).then(data => {
+    let url = 'http://www.espruino.com/webide?code=' + encodeURIComponent(data);
+    window.open(url, '_window');
+  });
+}
+
+function onGattDisconnected(evt) {
+  const device = evt.target;
+  logInfo(`Disconnected from GATT on device ${device.name}.`);
+}
+
+async function startTest() {
+  clearStatus();
```

Comment:
I noticed while testing this manually that `clearStatus()` doesn't reset `numErrors` and so if a previous test run failed a successful run will still be marked as "FAIL".

Reply:
Thx for catching that. Fixed.